subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/InputFileChanges.java

```diff
@@ -16,9 +16,23 @@
 
 package org.gradle.internal.execution.history.changes;
 
+import org.gradle.api.InvalidUserDataException;
 import org.gradle.internal.change.ChangeContainer;
 import org.gradle.internal.change.ChangeVisitor;
 
 public interface InputFileChanges extends ChangeContainer {
     boolean accept(String propertyName, ChangeVisitor visitor);
+
+    InputFileChanges EMPTY = new InputFileChanges() {
+
+        @Override
+        public boolean accept(ChangeVisitor visitor) {
+            return true;
+        }
+
+        @Override
+        public boolean accept(String propertyName, ChangeVisitor visitor) {
+            throw new InvalidUserDataException("Cannot query incremental changes: No incremental properties declared.");
```

Comment:
We should probably report the property name here.

Reply:
Done.